src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/CustomAttributeDecoder.cs

```diff
@@ -60,12 +68,32 @@ public CustomAttributeValue<TType> DecodeValue(EntityHandle constructor, BlobHan
                 throw new BadImageFormatException();
             }
 
-            ImmutableArray<CustomAttributeTypedArgument<TType>> fixedArguments = DecodeFixedArguments(ref signatureReader, ref valueReader, parameterCount);
+            BlobReader genericContextReader = default;
+            if (!attributeOwningTypeSpec.IsNil)
+            {
+                // If this is a generic attribute, grab the instantiation arguments so that we can
+                // interpret the constructor signature, should it refer to the generic context.
+                genericContextReader = _reader.GetBlobReader(attributeOwningTypeSpec);
+                if (genericContextReader.ReadSignatureTypeCode() == SignatureTypeCode.GenericTypeInstance)
```

Comment:
>if (genericContextReader.ReadSignatureTypeCode() == SignatureTypeCode.GenericTypeInstance [](http://example.com/codeflow?start=16&length=89)

Shouldn't this set attributeOwningTypeSpec to nil if the type code is not GenericTypeInstance?